decentraland/components/presentational/BuyLand.js

```diff
@@ -1,38 +1,60 @@
 import React from "react";
-import { StyleSheet, View } from "react-native";
+import { StyleSheet, View, Text } from "react-native";
 import { responsiveHeight } from "react-native-responsive-dimensions";
 import PropTypes from "prop-types";
 import LandForSale from "./LandForSale";
 import Button from "./Button";
 import Colors from "@constants/Colors";
 
 export default function BuyLand(props) {
-  const { landForSale, onBuy, waitingForAccountSetup } = props;
+  const { landForSale, onBuy, accountInfo } = props;
+  const {
+    account,
+    setupInProgress: waitingForAccountSetup,
+    fundedWithEthers,
+    fundedWithMana,
+    approvedMarketplace,
+  } = accountInfo;
+
+  let waitingMessage = "";
+  if (waitingForAccountSetup) {
+    if (account == null) waitingMessage = "Generating account...";
+    else if (!fundedWithEthers) waitingMessage = "Funding account with ETH...";
+    else if (!fundedWithMana && !approvedMarketplace)
```

Comment:
@marcelomorgado I think it may end up making our code easy to reason about if we move a single enumerable list of strings piece of state through states rather than dealing with the combinatorial complexity of all the booleans

Reply:
Hmm, create a new state var with the message and keep the booleans or delete the booleans?
I've created these flags thinking on the future screens with checks for each step.

Reply:
The enumerable list could be generated from the booleans still, but then when showing the messages we could use the enumerable list var

Reply:
Actually I take that back - I think the best options are either to store state in redux as the enumerable list (my favorite option) or keep it as-is.

Reply:
That makes sense?
```javascript
accountCreationStatus = ['NONE','GENERATING_ACCOUNT', 
'FUNDING_WITH_ETH', 
'FUNDING_WITH_MANA_AND_APPROVING_MARKETPLACE',
'FUNDING_WITH_MANA',
'APPROVING_MARKETPLACE',
'READY_TO_USE']
```

Reply:
Yeah, and `FUNDING_WITH_MANA_AND_APPROVING_MARKETPLACE` can transition to `FUNDING_WITH_MANA` or `APPROVING_MARKETPLACE`? That sounds great! I think this will be a big improvement in "easiness to reason about and ability to avoid bugs"

Reply:
And to clarify, there would be no more account creation status booleans in redux after making this change, right? Just this one var?

Reply:
> Yeah, and `FUNDING_WITH_MANA_AND_APPROVING_MARKETPLACE` can transition to `FUNDING_WITH_MANA` or `APPROVING_MARKETPLACE`? That sounds great! I think this will be a big improvement in "easiness to reason about and ability to avoid bugs"

We can pick one of both this big state is to cover the parallelism.

Reply:
> And to clarify, there would be no more account creation status booleans in redux after making this change, right? Just this one var?

Yes, it can be done (only the ENUM state) for sure.

Reply:
> We can pick one of both this big state is to cover the parallelism.

Yep, good call.

> Yes, it can be done (only the ENUM state) for sure.

Okay, awesome!

Reply:
Should we resolve this?

Reply:
I'll do this change yet.

Reply:
So, to clarify, 
1. We'll move all these props `setupInProgress,fundedWithEthers,fundedWithMana,marketplaceApproved` to `creationStatus` that will be a enum of strings.
and
2. For the paralellism, we'll assume that the `Mana funding` step occurs before `Marketplace approval` for the user feedback.

It these right?

Reply:
I agree with 1!

As for 2:

> For the parallelism, we'll assume that the `Mana funding` step occurs before `Marketplace approval` for the user feedback.

I think for user feedback we can trigger either state transition in redux and show whichever message appears first.

Reply:
> We can pick one of both this big state is to cover the parallelism.

Oh, maybe I didn't understand this point. I don't think I agree - I'll try to diagram this.

Reply:
Oh nevermind, I do agree that's the way to go for now. I'll show a diagram of how I think it can work once we're using event listeners in addition to shipping the two transactions in parallel

Reply:
![Frame](https://user-images.githubusercontent.com/2731712/55254967-4f59d400-5227-11e9-832c-937854575c80.jpg)

Reply:
Got it. I've just committed the redux state change.

Reply:
Oh looking at your code, I see how even without events we still are able to react to each of the two tx done in parallel separately, so it is like the diagram above, right?

Reply:
I think we can resolve this because it's now like the diagram above

Reply:
Yep

Reply:
Thanks. Resolving